state.go

```diff
@@ -308,12 +307,6 @@ func (a *Autopilot) buildServerState(inputs *nextStateInputs, srv raft.Server) S
 		state.State = RaftNone
 	}
 
-	// overwrite the raft state to mark the leader as such instead of just
-	// a regular voter
-	if srv.ID == inputs.LeaderID {
```

Comment:
This was moved a little lower to so that setting the `IsLeader` field could be fixed up if the application's `KnownServers` callback doesn't set it. Setting that other field would be overwritten by the code between here and there.